crates/docs/src/lib.rs

```diff
@@ -36,25 +36,50 @@ pub fn generate_docs_html(root_file: PathBuf) {
     fs::create_dir_all(build_dir).expect("TODO gracefully handle being unable to create build dir");
 
     // Copy over the assets
-    fs::write(
-        build_dir.join("search.js"),
-        include_str!("./static/search.js"),
-    )
-    .expect("TODO gracefully handle failing to make the search javascript");
-
-    fs::write(
-        build_dir.join("styles.css"),
-        include_str!("./static/styles.css"),
-    )
-    .expect("TODO gracefully handle failing to make the stylesheet");
-
-    fs::write(
-        build_dir.join("favicon.svg"),
-        include_str!("./static/favicon.svg"),
-    )
-    .expect("TODO gracefully handle failing to make the favicon");
-
-    let template_html = include_str!("./static/index.html")
+    // For debug builds, read assets from fs to speed up build
+    // Otherwise, include as string literal
+
+    // Copy search.js
+    #[cfg(not(debug_assertions))]
+    let search_js = include_str!("./static/search.js");
+
+    #[cfg(debug_assertions)]
+    let search_js = fs::read_to_string("./crates/docs/src/static/search.js").unwrap();
+
+    fs::write(build_dir.join("search.js"), search_js)
+        .expect("TODO gracefully handle failing to make the search javascript");
+
+    // Copy styles.css
+    #[cfg(not(debug_assertions))]
+    let styles_css = include_str!("./static/styles.css");
+
+    #[cfg(debug_assertions)]
+    let styles_css = fs::read_to_string("./crates/docs/src/static/styles.css").unwrap();
+
+    fs::write(build_dir.join("styles.css"), styles_css)
+        .expect("TODO gracefully handle failing to make the stylesheet");
```

Comment:
I think this is fine, since this only applies to debug builds - I'd just have it panic with something like "tried to write `styles.css` to _________ but got this error: ___________"

Reply:
Just to clarify, would you like me to update the error handling for the `fs::read_to_string` call, the `fs::write` call, or both? 

Or is there a nice way to handle both errors at once without over-engineering it? (Sorry, my limited Rust knowledge is showing.)

Reply:
Actually `.unwrap()` is probably fine for all the error handling here. 😄 

I just saw you asked for feedback on error handling, so the main point I wanted to get across is that panicking is fine for error handling here since it only comes up in dev builds!

Reply:
Yeah, definitely makes sense that we don't need to spend as much time on error handling for debug builds!

I think the `fs::write` calls happen for both debug and release builds though? (These calls were already here, and I just changed the `contents` argument.)

Reply:
Oh gotcha - then yeah, no need to change anything else about them unless you'd like to. 👍

Reply:
I went ahead and added some slightly nicer errors, since I could! There are still places in the file we could add better errors though.